### ISSUE TITLE
Update RHVoice GitHub version to match Gplay store version

### ISSUE THREAD

**Issue by loyl1** [RHVoice/RHVoice#677]
The GitHub version of RHVoice lags 5 months behind the Gplay store version. Please update the apk

**Comment by zstanecic** [comment]
Hi,

Github is no longer the reference for new gogle play store builds, as it is hard to maintain across the language versions.

Please visit the main project site:

https://www.rhvoice.org

to find the newest apk.

 

 

From: loyl1 ***@***.***> 
Sent: Tuesday, January 10, 2023 10:57 AM
To: RHVoice/RHVoice ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [RHVoice/RHVoice] Update RHVoice GitHub version to match Gplay store version (Issue #677)

 

The GitHub version of RHVoice lags 5 months behind the Gplay store version. Please update the apk

**Comment by pm4rcin** [comment]
What about f-droid version? Maybe just create tag on the commit you feel comfortable and the rest could be done on f-droid's side?

**Comment by nicoursi** [comment]
> Github is no longer the reference for new gogle play store builds, as it is hard to maintain across the language versions.

Does this mean RH voice is no longer opensource?  If it is not the case,  where can the source code for the newest version be found?

If there's non code available,  then your website is misleading and the word "open source" should be removed from the website.

**Comment by zstanecic** [comment]
The project is open-source, but we have submodules, and the project is robust.

**Comment by licaon-kter** [comment]
@zstanecic can you rephrase that in relation to tagging a release on Github so F-Droid can pick it up and build?

**Comment by TheLastProject** [comment]
For context, [the F-Droid build of RHVoice is currently set up to read the fdroid.txt files of the latest GitHub release](https://gitlab.com/fdroid/fdroiddata/-/blob/f0b5865fd1bb81bb8240e576028132242b6d7d2f/metadata/com.github.olga_yakovleva.rhvoice.android.yml#L70).

Because the latest release on GitHub is [1.8.0](https://github.com/RHVoice/RHVoice/releases/tag/1.8.0), F-Droid doesn't know newer versions exist.

Creating a new tag and adding an fdroid.txt file to it will allow F-Droid to know a new version is available and will publish it.

So it would be awesome if you could create a git tag and add an fdroid.txt file to it so the F-Droid version of RHVoice can be updated. Thank you :+1:

**Comment by pm4rcin** [comment]
@alex19EP @Olga-Yakovleva or whoever makes releases for google play could you take a look at that issue? We would really appreciate it since it doesn't require much from your side. Thanks in advance! :)

**Comment by licaon-kter** [comment]
Update manually as upstream does not seem to care :(

https://gitlab.com/fdroid/fdroiddata/-/commit/5cb949261900e5bd653214f300e4dd5221590b52

**Comment by pm4rcin** [comment]
@licaon-kter the new version 2.0.30 is on google play. Could you update once again? :)

**Comment by licaon-kter** [comment]
@pm4rcin not really as https://github.com/RHVoice/RHVoice/tree/51759044ff1871310fc222e21c5655f960ebfca3/src/android/RHVoice-core#L3-L6 were not updated....

**Comment by pm4rcin** [comment]
@zstanecic could you or someone else like @Olga-Yakovleva update the strings in build.gradle as licaon said so f-droid could build new version? In google play the version is newer so I guess you have changed this file there.

**Comment by pm4rcin** [comment]
@licaon-kter looks like the version in `build.gradle` was updated as you said. Could you push it to f-droid? :)

**Comment by licaon-kter** [comment]
@pm4rcin done https://gitlab.com/fdroid/fdroiddata/-/commit/de8d277be9e1df482157b1ffc8107180c7cc125b

**Comment by licaon-kter** [comment]
Which version was published on Play on Apr 07?

**Comment by licaon-kter** [comment]
Or May 01?

**Comment by foolishgrunt** [comment]
@licaon-kter 

As of August 7, the latest GPlay version is 2.8.1

**Comment by pm4rcin** [comment]
> @licaon-kter
> 
> As of August 7, the latest GPlay version is 2.8.1

Doesn't matter really because it's just a number. The recent commit in this repo changed version so it's up to date.

**Comment by licaon-kter** [comment]
Latest F-Droid is 1.16.3 and it's up to date for now, https://github.com/RHVoice/RHVoice/commit/dbb0fb59107357ce384cf135816e21c6956314b6

**Comment by licaon-kter** [comment]
can we get a version bump? latest Play is from July 20

**Comment by zstanecic** [comment]
@licaon-kter Just yesterday, I have bumped the tag to 1.18.1, The Fdroid app should be updated. Is there anything i need to do?

**Comment by licaon-kter** [comment]
thanks

**Comment by zstanecic** [comment]
Tell me, what should be done to update the fdroid app, or you will be on charge of this @licaon-kter

**Comment by licaon-kter** [comment]
already done @zstanecic https://gitlab.com/fdroid/fdroiddata/-/commit/e2f7c9ebb84c3fb2fc725a901ffac0b4050c61aa

any chance you can put versionName and code in a file so we can enable autoupdates?

Eg. in the past you had fdroid.txt like https://github.com/RHVoice/RHVoice/releases/tag/1.8.0

**Comment by zstanecic** [comment]
Yes, I can do it. I will need your assistance in order to verify the correctness.

**Comment by zstanecic** [comment]
@licaon-kter Done. fdroid.txt is in the release asset.
https://github.com/RHVoice/RHVoice/releases/download/1.18.1/fdroid.txt

**Comment by zstanecic** [comment]
Update 2: @licaon-kter just updated the version code as in your diff.

**Comment by licaon-kter** [comment]
@zstanecic can you make case consistent between the lines? :)

**Comment by zstanecic** [comment]
@licaon-kter Sure, just corrected.

**Comment by licaon-kter** [comment]
thanks, fyi https://gitlab.com/fdroid/fdroiddata/-/commit/03d0ab87f5fc412e8987c3b42cd1c42ff720009e

**Comment by zstanecic** [comment]
hi @licaon-kter,
Thanks. What is the procedure now? the tests pass, but we don't have it in the app

**Comment by licaon-kter** [comment]
https://gitlab.com/fdroid/wiki/-/wikis/FAQ#how-long-does-it-take-for-my-app-to-show-up-on-website-and-client

**Comment by zstanecic** [comment]
Update: @licaon-kter,
The version 1.18.2 is just tagged with a small fix: can you update the metadata according to 1.18.2 fdroid.txt?

**Comment by licaon-kter** [comment]
autoupdate works, so it would have been picked up in about 12h anyway 😉 

but opened a MR here: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/34592

**Comment by zstanecic** [comment]
@licaon-kter thanks!

**Comment by zstanecic** [comment]
Hi @licaon-kter again,
Maybe you know... The reproducibility status for RHVoice 1.18.2 is not shown. Is there something that we missed? The last reproducible build is 1.16.050

**Comment by licaon-kter** [comment]
> again,

as said, autoupdate works, it will be picked up in about 18h :)

> The last reproducible build is 1.16.050

the verification server is behind schedule, it takes time

**Comment by zstanecic** [comment]
Thanks for explanation @licaon-kter!
b

**Comment by zstanecic** [comment]
Because Fdroid builds are automatic, Closing this issue.

**Comment by loyl1** [comment]
The rhvoice.org website referenced as the source of current builds, for android points exclusively to Google play. I do see now that there is an F-Droid option again (from posts here) which helps for those of us who don't use Google Play. But not ideal as it adds another party into the signing and trust chain

**Comment by zstanecic** [comment]
@loyl1 But, at least the two builds are legit, and you can see it from both places.

**Comment by licaon-kter** [comment]
@zstanecic https://gitlab.com/fdroid/checkupdates-bot-fdroiddata/-/jobs/13736567197#L2782 ?

**Comment by zstanecic** [comment]
Will fix that soon. Can I reuse the taga and version code?

**Comment by licaon-kter** [comment]
As you want...

**Comment by zstanecic** [comment]
Thanks.

**Comment by zstanecic** [comment]
@licaon-kter fixed please re-run the checks

**Comment by licaon-kter** [comment]
thanks

**Comment by zstanecic** [comment]
Hi @licaon-kter can you pm me on my e-mail visible in my profile? I need to ask you one very important question.

**Comment by licaon-kter** [comment]
just email us at team at f-droid.org @zstanecic  😉

**Comment by zstanecic** [comment]
@licaon-kter done.
